engine/src/main/java/com/sandboni/core/engine/finder/bcel/visitors/annotations/SuiteAnnotationProcessor.java

```diff
@@ -0,0 +1,42 @@
+package com.sandboni.core.engine.finder.bcel.visitors.annotations;
+
+import com.sandboni.core.engine.finder.bcel.visitors.AnnotationUtils;
+import com.sandboni.core.engine.finder.bcel.visitors.Annotations;
+import com.sandboni.core.engine.sta.Context;
+import com.sandboni.core.engine.sta.graph.LinkFactory;
+import com.sandboni.core.engine.sta.graph.LinkType;
+import com.sandboni.core.engine.sta.graph.vertex.TestSuiteVertex;
+import org.apache.bcel.classfile.AnnotationEntry;
+import org.apache.bcel.classfile.JavaClass;
+
+import java.util.Arrays;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import static com.sandboni.core.engine.finder.bcel.visitors.AnnotationUtils.getAnnotationParameter;
+import static com.sandboni.core.engine.sta.graph.vertex.VertexInitTypes.TEST_SUITE_VERTEX;
+
+public class SuiteAnnotationProcessor implements RunWithAnnotationProcessor {
+    private static final String VALUE = "value";
+    private boolean isSuite;
```

Comment:
Multiple executions of Sandboni run concurrently in the same JVM when Gradle parallel builds are enabled. As you are creating a static map in the Factory, a single instance of SuiteAnnotationProcessor is shared across multiple Threads and this class is not ThreadSafe due this property "isSuite". As a suggestion, you could:
1. Find another approach for isSuite. Would  be a cleaner API approach but maybe refactoring is expensive.
2. Instead of create a Map in the factory with a single shared instance, create a Map<Key, Supplier<Instance>> and create a new instance in each invocation to this factory.
Feel free to implement another solution :)

Reply:
FYI visitJavaClass is synchronized, but you are right it's not the best implementation for thread-safety. I will change it and you tell me what you think.

Reply:
Fixed. please have a look thanks.